### PR TITLE
Add max_characters to rules

### PR DESCRIPTION
**Pull request by HarikalarKutusu**
Implements #182

### CODE REVIEW

File: src/checker.rs [hunk 1]
```diff
@@ -3,6 +3,11 @@ use toml::Value;
 use regex::Regex;
 
 pub fn check(rules: &Rules, raw: &str) -> bool {
+
+    fn in_limit(x: usize, min_val: usize, max_val: usize) -> bool {
+        return (x >= min_val) && (x <= max_val)
```
Comment (MichaelKohler) [inline src/checker.rs:8]: Minor: could you move this outside the function?

File: src/rules.rs [hunk 2]
```diff
@@ -66,6 +67,7 @@ impl Default for Rules {
             min_word_count: 1,
             max_word_count: 14,
             min_characters: 0,
+            max_characters: std::usize::MAX,
```
Comment (MichaelKohler) [inline src/rules.rs:70]: `std::usize::MAX` is deprecated, could you replace it with `usize::MAX`? https://doc.rust-lang.org/std/primitive.usize.html#associatedconstant.MAX

File: src/rules.rs [hunk 3]
```diff
@@ -100,6 +102,7 @@ mod test {
         assert_eq!(rules.min_word_count, 1);
         assert_eq!(rules.max_word_count, 14);
         assert_eq!(rules.min_characters, 0);
+        assert_eq!(rules.max_characters, 999);
```
Comment (MichaelKohler) [inline src/rules.rs:105]: Where is 999 coming from? I think the tests failed before so you didn't get an automated comment here. This should be `usize::MAX`.

File: src/checker.rs [hunk 4]
```diff
@@ -12,7 +17,7 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
                 .nth(1)
                 .map(|c| !c.is_alphabetic())
                 .unwrap_or_default()
-        || trimmed.chars().filter(|c| c.is_alphabetic()).count() < rules.min_characters
+        || !in_limit(trimmed.chars().filter(|c| c.is_alphabetic()).count(), rules.min_characters, rules.max_characters)
```
Comment (MichaelKohler) [inline src/checker.rs:20]: To make this line a bit shorter, what do you think about assigning the count to a variable before this fill check?

File: src/checker.rs [hunk 5]
```diff
@@ -139,6 +144,17 @@ mod test {
...
 
+    #[test]
+    fn test_max_characters() {
+        let rules : Rules = Rules {
+            max_characters: 25,
+            ..Default::default()
+        };
+
+        assert!(!check(&rules, &String::from("not less than or equal to 25")));
+        assert!(check(&rules, &String::from("less than or equal to 25")));
```
Comment (MichaelKohler) [inline src/checker.rs:155]: I'm not seeing anything obviously wrong here, can you have a deeper look into why this test case fails? Might also be some default rule?

**Review by MichaelKohler — CHANGES_REQUESTED** [review]
Thank you! Generally this looks good to me, I've left a few comments.

File: src/rules.rs (hunk 2 above)
Comment (HarikalarKutusu) [reply src/rules.rs:70]: Yep just saw that and already replaced it everywhere.

File: src/checker.rs (hunk 4 above)
Comment (HarikalarKutusu) [reply src/checker.rs:20]: Yep, better...

File: src/checker.rs (hunk 5 above)
Comment (HarikalarKutusu) [reply src/checker.rs:155]: I'm not sure either. I was playing with MAX and 999 but could not pass the tests yet (my first lines of code in Rust).

File: src/rules.rs (hunk 2 above)
Comment (HarikalarKutusu) [reply src/rules.rs]: Done...

File: src/checker.rs (hunk 4 above)
Comment (HarikalarKutusu) [reply src/checker.rs]: Done...

File: src/checker.rs (hunk 5 above)
Comment (HarikalarKutusu) [reply src/checker.rs]: Let's see if it is fixed here also.

**Review by MichaelKohler — APPROVED** [review]
Thanks for the other `usize::MAX` adjustments. All looking good to me now. Thanks, well done on your first Rust contribution!

**Pull request merged**